developer_guide.md

```diff
@@ -0,0 +1,214 @@
+# Lego 3 - Developer Guide
+
+This file has 2 main purposes:
+1. Help you develop Lego 3 (core, libs or tests), avoid you from common mistakes and save your time when entering the project.
+2. Keep Lego 3 repository clean and order, avoid conflicts between developers and save the time of who reads/uses your code. 
+
+## Background
+
+Lego 3 is a new distributed testing infrastructure used to test and monitor our products in the network.
+
+
+## Required Knowledge
+
+### Project
+
+#### Requirements
+
+*Optional*
+
+Lego 3 is a completely new project, written from scratch, but its requirements based on insights from the previous testing infrastructure we used.
+
+Read the requirements for this project [here](https://docs.google.com/document/d/1gSl9_jS_pIAkGtNLJSBKUGRbcOhALKMzv1ewWZSSElg/edit?usp=sharing) under "Requirments".
```

Comment:
The link here is with editing permissions and this repo is public!!!
Revoke the link, delete the link and preferably delete the link from history.
Fix all the links

Reply:
Check again

Reply:
Check again